### PR TITLE
Add logging to tar, fix extract when parent dir is .

### PR DESCRIPTION
**Pull request by csrwng**
Adds additional methods to the tar interface to log the files being extracted/archived (similar to -v flag on the tar command). 
Also fixes name truncation when the parent directory is the current directory.

### CODE REVIEW

**Comment by csrwng** [comment]
@bparees or @gabemontero ptal

**Comment by csrwng** [comment]
[test]

File: pkg/tar/tar.go [hunk 1]
```diff
@@ -45,6 +51,12 @@ type Tar interface {
 	// Times out if reading from the stream for any given file
 	// exceeds the value of timeout
 	ExtractTarStream(dir string, reader io.Reader) error
+
+	// ExtractTarStreamWithLogging extracts files from a given tar stream.
+	// Times out if reading from the stream for any given file
+	// exceeds the value of timeout.
```
Comment (bparees) [inline pkg/tar/tar.go]: the value of what timeout?

**Comment by bparees** [comment]
lgtm [merge]

File: pkg/tar/tar.go (hunk 1 above)
Comment (csrwng) [inline pkg/tar/tar.go]: should be defaultTimeout in this file

**Pull request merged**